pkg/azure/provider_test.go

```diff
@@ -120,3 +121,179 @@ func TestParseAzureEnvironment(t *testing.T) {
+CwUAA4ICAQDC1dMw07Ooec28Td1H+oVLel/P5uvtctRz54JL4HHMxrlSxLqIvrye
+cnediGSCxFSrSvg9sUfq6UILcF8KUYmmTjsXQC6QMVa7Kt5BggLI+BS5Vlj6BEQp
+2V5EtXXnU8YIHB7gt5iZU07Yk3aBQioGgSUKOEMjlarnHV50XB6NkS9x3MO+pw/Y
+jE1DxmRNKHXOyOgv4PQbqPyXFqryL2kq6QpDqqj6ucH/49IjqNVf7dVJPl3sC13v
+wVDGVlznJWRbU1WyQpkFGFaPVwqf7e7OWk0pxW+0RptzcAjZQa5qGlxXPF4H0v16
+sSDcI7qPcRj8ztL/Uw2r9q/n5DY+ag5IEB4Acw5nJl6LJWcFdYb3YAfZWPwVzOqy
+iUU/Lli9D/Jsmb7erXT3AFD0L4JgRnrLQiPqjNfIsAWsVHEuCzPwQEd7Tvi127Di
+wSIW87flu9GeZbheLPeXzQmGQjgGT8rvX7nXtSEsSKhaefHwDPfxFH33pRFc7YUm
+GsmWcS0+ewWlzF1d/m3ist0C4w8GyQ5jYuamyErVNa/pQ1iHsi7A9wfq2XrUabi6
+a5H8yLyBQ0iqu9hEcBHTmlD6rupWVU3MRee7H+956J5rbzJkngils4TP78jbgQIO
+fij5z2FZL+lrHgkj7cE29O8plJNbHTi+AsguKCy4OSBSha53tFUB6A==
+-----END CERTIFICATE-----
+`,
+			expectedKey: `-----BEGIN PRIVATE KEY-----
+MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQCeLkx+0iAqaiF/
+Ts9H+RJh20+vjfZ5f2CkeKCWlaQjKgIMa2y35MvqKRimqV70UjY2cTiuUpKl8fWo
+mmBGpSoSsQzC647qZFCc1KuzLII86OvuwYAPMU+vpIQfJ5/b5m0dGMclMF12KH4R
+aONdE7v8waHsoAWWnRY9ogmM9J5DW9Y0fevDCcnXC57PQdV+xcDcJBGQS4DHfekP
+xkiKXSiUlLtodV6hRbLFnQb+4Nmdbqaqxw4hSx+3bTduduruhPSYuONCOAImuiFH
+cB65L55sC+irKYImK/grEItgOl0PNU0fsJCW68g6fWw+OX2gdVpv9gazLc3dFGgQ
+WbMRBoPJAgMBAAECggEAdICrLJj8weCLLWXROzRSeqp5pVZM262ac2i49k/epVuS
+SK1yVHr6SUvdcRq76f0HGtAHLDV69ygfg/+3uzB4rW2jdYjjKPWbffcRQxUcE1qG
+MsJn3Ei4ZSgpe3zGu8KaSNzwCA74N0eZmp8DHpGjWoUPCuXNV+H4+In4uM2OJxM7
+VcqlTXPdp94XCOYM9K138fSoegwpu+ef37fm+tSR5rIyh+pqi5SCiSKn5R4UVRCu
+dXEYclxL603tt1eYl/C8MqmtrVCcReSlpfLUX9xuT145egz2rPfMV6Bl1+6GxfJn
+4PZOnbQZXXgo8UDgwlqhZHKll2t26awXwyTja0VDkQKBgQDNFMKazHlOSexqnuBL
+HOVE/sdA3GMtlGlwGjhYxSRqoz8KBag7RVDWkrfgXbHX+G4aJc3PErz+aKxSV0XA
+DG9WjJDJOojsFT9/XWpCu1aA26lT1QJL6yDcLdokjQC586JwhkV7JR2YNkRduF9H
+NnqcA/iD9y2oocn6UnRyo+VQFQKBgQDFdH5+c6Iceup7LuVDBhBXH9RbOVqEqSA3
+QvTe5U46cul7bSjYgQn+ksnn9UML39sytnVrnC3LOLZmIEJzy1n22rY0b82KrnCY
+XU9sjGeniIBG8j0fXKDvSGBsy5NL1u7zAU3v9sCnArbACFPVu7S/hO7V24ZiJJHt
+/hDVcOWd5QKBgCDkNn37U21R/9/t0U1aug7ByhVGA4YY6nw3SFg8biXIPuENnTi8
+WkW/zEvo2xAnYQlCjOqsN7GZ+iFOq/osRGMeMk6D29f5ZHC5+8PuJeaO1G6EmFSy
+xldp5zW7g6VPRPtFHbmtbzytX3OkkWtremixXldT+ne0Ux+Zv+FvFeUtAoGAFPg6
+NtOw87VaEZr5XhTWx2np84YzxsLvWO8TcliH5k0t3p3JKLULiq2sI6Y4aJptfQVD
+kxoTAvIS7OWgKQv/kefIUelNutyruIKwXKbMm04z0VUIiLwdm0vkcaltCzDYT5Zj
+4IgkDZiML/iybpBwsaY8dxnJO8MGfG/u+bvzpsECgYEAirskIr2+Gn2IwFnLf14g
+I6+joeBo7bu/31yhLm4XcDPD2rl3PMIPCvxRe5FfPQOIAF9Vxibsomdhi30Z2Vmo
+6GxVnQB9Zvy+V0LkfHvF7qUyxcY76o7SzwrZ0e4or3snPLMyOV8ZIH0wfolqHm9y
+Uz7sJMWoq7mOrINHQ0ZmaiE=
+-----END PRIVATE KEY-----
+`,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			content, err := decodePKCS12(tc.value, false, true)
+			if err != nil {
+				t.Fatalf("expected nil err, got: %v", err)
+			}
+			if content != tc.expectedCert {
+				t.Fatalf("certificate mismatch")
+			}
+			content, err = decodePKCS12(tc.value, true, false)
+			if err != nil {
+				t.Fatalf("expected nil err, got: %v", err)
+			}
+			fmt.Println(content)
```

Comment:
Do we need this still? Assume this is for debug purpose?

Reply:
Removed